React/CoreModules/RCTAlertController.m

```diff
@@ -35,6 +35,7 @@ - (void)show:(BOOL)animated completion:(void (^)(void))completion
 
 - (void)hide
 {
+  [_alertWindow setHidden: YES];
```

Comment:
[_alertWindow removeFromSuperview]; feels more correct to me - have you tried that?

Reply:
>If the view’s superview is not nil, the superview releases the view.

What's the superview of a `UIWindow`?

Reply:
The correct way to remove a `UIWindow` in iOS is to set its hidden property to `YES`.
Also, needs to remove all references to the window and ARC will automatically free this `UIWindow`.

The line after this change, set the `_alertWindow` reference to `nil`, but the window is already associated with a scene (see the screenshots from this PR). So we also need to remove the `windowScene` from that window, as recommended by Apple: https://developer.apple.com/documentation/uikit/uiwindowscene/3198091-windows.
>To remove the window from the current scene, or move it to a different scene, change the value of the window's windowScene property.

Reply:
@paddlefish since I can't push this change to your branch, I created another PR based on your changes and my comment and co-authorized you. @philIip & @lunaleaps you are welcome to take a look: https://github.com/facebook/react-native/pull/32833.

Reply:
@asafkorem why wouldn't the alertWindow have a superview?

Reply:
> @asafkorem why wouldn't the alertWindow have a superview?

@philIip the `_alertWindow` is a `UIWindow`, which doesn't have any superview by default, and there isn't any code where we define it as a subview of another view. Therefore, calling `removeFromSuperview` will do nothing, as the doc says (https://github.com/facebook/react-native/pull/32305#discussion_r778885998).

> i'm not convinced that setting it to hidden is really related to this core issue

Setting the alert's window to be hidden from the `hide` method ensures that this UI operation is handled correctly, **regardless of the memory management**.

Removing any strong reference (made by this class) to this object is done to ensure that this window will be released from memory **once no other object is pointing at it**.
Theoretically, this window could be retained from another place, for example some object that kept a strong reference to the topmost visible window ATM this window was presented, or some object that keeps a strong reference to the generated alert view-controller (that has a strong reference to this window). In these examples, the window will still be stuck after calling `hide` until its reference count will be zero (same as the original bug).

Reply:
@asafkorem I understand, where i'm coming from is that it is not impossible for UIWindow to have a superview (it is a UIView after all) since we can't control or know what UIKit does under the hood with the UIWindow management during runtime, and that could be a lingering strong reference. It's unlikely, but it's a possibility as well since the API would allow it.

Reply:
@philIip that's right, it is possible by the API, but this should not happen in this case, since this class present and manage this window, and it does not do that.
If it happens, for any reason, it will indicate on some other bug that must be addressed, and I strongly recommend not to obscure this potential bug from this method.